wgpu/Cargo.toml

```diff
@@ -25,33 +25,78 @@ targets = [
 
 [features]
 default = ["wgsl", "dx12", "metal"]
-# Apply run-time checks, even in release builds. These are in addition
-# to the validation carried out at public APIs in all builds.
-strict_asserts = ["wgc?/strict_asserts", "wgt/strict_asserts"]
+
+#! ### Backends
+# --------------------------------------------------------------------
+#! ⚠️ WIP: Not all backends can be manually configured today.
+#! On Windows & Linux the Vulkan & GLES backends are always enabled.
+#! See [#3514](https://github.com/gfx-rs/wgpu/issues/3514) for more details.
+
+## Enables the DX12 backend on Windows.
+dx12 = ["wgc?/dx12"]
+
+## Enables the Metal backend on macOS & iOS.
+metal = ["wgc?/metal"]
+
+## Enables the GLES backend via [ANGLE](https://github.com/google/angle) on Windows & macOS using.
+angle = ["wgc?/gles"]
+
+## Enables the Vulkan backend on macOS & iOS.
+vulkan-portability = ["wgc?/vulkan"]
+
+## Enables the GLES backend on Wasm
+##
+## * ⚠️ WIP: Currently will also enable GLES dependencies on any other targets.
+## * ⚠️ WIP: This automatically disables the WebGPU feature. See [#2804](https://github.com/gfx-rs/wgpu/issues/3514).
+webgl = ["hal", "wgc/gles"]
+
+#! ### Shading language support
+# --------------------------------------------------------------------
+#! The following features do not have any effect on the WebGPU backend.
+
+## Enable accepting SPIR-V shaders as input.
 spirv = ["naga/spv-in"]
+
+## Enable accepting GLSL shaders as input.
 glsl = ["naga/glsl-in"]
+
+## Enable accepting WGSL shaders as input.
 wgsl = ["wgc?/wgsl"]
+
+#! ### Logging & Tracing
+# --------------------------------------------------------------------
+#! The following features do not have any effect on the WebGPU backend.
+
+## Apply run-time checks, even in release builds. These are in addition
+## to the validation carried out at public APIs in all builds.
+strict_asserts = ["wgc?/strict_asserts", "wgt/strict_asserts"]
+
+## Log all API entry points at info instead of trace level.
+api_log_info = ["wgc/api_log_info"]
+
+## Allow writing of trace capture files.
+## See [`Adapter::request_device`].
 trace = ["serde", "wgc/trace"]
+
+## Allow deserializing of trace capture files that were written with the `trace` feature.
+## To replay a trace file use the [wgpu player](https://github.com/gfx-rs/wgpu/tree/trunk/player).
 replay = ["serde", "wgc/replay"]
-# Implement `Send` and `Sync` on Wasm.
+
+#! ### Other
+# --------------------------------------------------------------------
+
+## Implement `Send` and `Sync` on Wasm, but only if atomics are not enabled.
+##
+## WebGL/WebGPU objects can not be shared between threads.
+## However, it can be useful to artificially enforce `Send` and `Sync`
+## anyways to make it easier to write cross-platform code.
+## This is technically *very* unsafe, but as long as objects can not be shared
+## with another thread it's fine, thus the restriction on having atomics disabled.
```

Comment:
```suggestion
## WebGL/WebGPU objects can not be shared between threads.
## However, it can be useful to artificially mark them as `Send` and `Sync`
## anyways to make it easier to write cross-platform code.
## This is technically *very* unsafe in a multithreaded environment, but on a wasm binary compiled without atomics
## we know we are definitely not in a multithreaded environment. 
```